File: code/frontend/src/utils/challenge.ts
export function difficultyClass(difficulty: string): string {
  const map: Record<string, string> = {
    beginner: 'difficulty-chip difficulty-chip--beginner',
    easy: 'difficulty-chip difficulty-chip--easy',
    medium: 'difficulty-chip difficulty-chip--medium',
    hard: 'difficulty-chip difficulty-chip--hard',
    insane: 'difficulty-chip difficulty-chip--insane',
  }
  return map[difficulty] || 'difficulty-chip difficulty-chip--unknown'
}

export function difficultyLabel(difficulty: string): string {
  const map: Record<string, string> = {
    beginner: '入门',
    easy: '简单',
    medium: '中等',
    hard: '困难',
    insane: '地狱',
  }
  return map[difficulty] || difficulty
}

File: code/backend/internal/shared/taxonomy/challenge.go
package taxonomy

const (
	DimensionWeb       = "web"
	DimensionPwn       = "pwn"
	DimensionReverse   = "reverse"
	DimensionCrypto    = "crypto"
	DimensionMisc      = "misc"
	DimensionForensics = "forensics"
)

var ValidDimensions = map[string]bool{
	DimensionWeb:       true,
	DimensionPwn:       true,
	DimensionReverse:   true,
	DimensionCrypto:    true,
	DimensionMisc:      true,
	DimensionForensics: true,
}

var AllDimensions = []string{
	DimensionWeb,
	DimensionPwn,
	DimensionReverse,
	DimensionCrypto,
	DimensionMisc,
	DimensionForensics,
}

const (
	DifficultyBeginner = "beginner"
	DifficultyEasy     = "easy"
	DifficultyMedium   = "medium"
	DifficultyHard     = "hard"
	DifficultyInsane   = "insane"
)

func IsValidDimension(dimension string) bool {
	return ValidDimensions[dimension]
}
